components/webui/imports/ui/SearchView/SearchResults.jsx

```diff
@@ -49,10 +49,15 @@ const SearchResults = ({
     const hasMoreResults = visibleSearchResultsLimit < numResultsOnServer;
 
     const handleLoadMoreResults = useCallback(() => {
-        setVisibleSearchResultsLimit(
-            (v) => (v + VISIBLE_RESULTS_LIMIT_INCREMENT)
-        );
-    }, []);
+        if (hasMoreResults) {
```

Comment:
Technically, if there weren't enough results to fill the page, the `setInterval` call would keep firing, right? Should we add an `if (false === hasMoreResults)` break out in the sensor's `useEffect`?

Reply:
Instead added the check inside the Intersection Observer callback so that the interval gets cleared when there are no `hasMoreResults`.

Reply:
Do we need it inside the IntersectionObserver? Wouldn't the flow go:

* hasMoreResults = true
  Add intersection observer and interval
* hasMoreResults = false
  Intersection observer and interval get cleared by useEffect's destructor
  useEffect runs adding the intersection observer but without the observer doing anything

Reply:
> Intersection observer and interval get cleared by useEffect's destructor

Right, I previously overlooked this part.